core/src/replay_stage.rs

```diff
@@ -328,10 +339,12 @@ impl ReplayStage {
         bank: &Bank,
         blocktree: &Blocktree,
         progress: &mut HashMap<u64, ForkProgress>,
-    ) -> Result<()> {
+    ) -> (Result<()>, Vec<Entry>) {
+        let mut entries = Vec::new();
         let result =
-            Self::load_blocktree_entries(bank, blocktree, progress).and_then(|(entries, num)| {
-                Self::replay_entries_into_bank(bank, entries, progress, num)
+            Self::load_blocktree_entries(bank, blocktree, progress).and_then(|(e, num)| {
+                entries = e;
+                Self::replay_entries_into_bank(bank, &entries, progress, num)
             });
```

Comment:
I think this would work too and avoids creating a `mut`
```rust
let (result, entries) =
    Self::load_blocktree_entries(bank, blocktree, progress).and_then(|(entries, num)| {
        (Self::replay_entries_into_bank(bank, &entries, progress, num), entries)
    });
```

Reply:
I was really hoping to find a more elegant way to write this! The problem is that I want to do two things, map the result and pull out the entries or default to an empty vec. Your suggestion won't compile because `and_then` has return type `Result` while you are returning a tuple.

Reply:
Wait the only consumer of `entries` is this line right?
```rust
tx_count += entries.iter().map(|e| e.transactions.len()).sum::<usize>();
```

We could just return the tx_count as `Result<(), u64>`.

Reply:
Do you mean a `(Result<()>, u64)`? Yeah, we can do that! I guess I would just worry that a u64 isn't self-documenting. What's the motivation to return a `u64` instead of a `Vec<Entry>`?

Reply:
> What's the motivation to return a u64 instead of a Vec<Entry>?

Just to reduce the API surface to the caller, as they currently only need the transaction count and not the full list of Entries.   Future refactoring is so much easier if the dependencies between all the things contain only the essentials